test/e2e/tests/test_rule_groups_namespace.py

```diff
@@ -0,0 +1,342 @@
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the Amazon Managed Prometheus (AMP) Rule Groups Namespace resource
+"""
+
+import logging
+import time
+import pytest
+
+from acktest.k8s import resource as k8s
+from acktest.k8s import condition
+from acktest import tags as tags
+from acktest.resources import random_suffix_name
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_prometheusservice_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+
+
+RESOURCE_KIND = "rulegroupsnamespace"
+RESOURCE_PLURAL = "rulegroupsnamespaces"
+
+
+MAX_WAIT_FOR_SYNCED_MINUTES = 10
+UPDATE_WAIT_AFTER_SECONDS = 5
+DELETE_WAIT_AFTER_SECONDS = 60
+CREATE_WAIT_AFTER_SECONDS = 90
+
+
+@service_marker
+@pytest.mark.canary
+class TestRuleGroupsNamespace:
+
+    def create_workspace(self, prometheusservice_client, workspace_alias) -> str:
+        try:
+            resp = prometheusservice_client.create_workspace(
+                alias=workspace_alias
+            )
+            workspace_id = resp['workspaceId']
+
+            return workspace_id
+
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+    def delete_workspace(self, prometheusservice_client, workspace_id):
+        try:
+            response = prometheusservice_client.delete_workspace(
+                workspaceId=workspace_id
+            )
+            return response
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+    def get_rule_groups_namespace(self, prometheusservice_client, workspace_id: str, name: str) -> dict:
+        try:
+            resp = prometheusservice_client.describe_rule_groups_namespace(
+                workspaceId=workspace_id,
+                name=name
+            )
+            return resp
+
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+
+    def assert_server_side_status(self, result_from_server, status):
+        assert 'status' in result_from_server['ruleGroupsNamespace']
+        assert 'statusCode' in result_from_server['ruleGroupsNamespace']['status']
+        assert result_from_server['ruleGroupsNamespace']['status']['statusCode'] == status
+
+
+    def test_successful_crud_rule_groups_namespace(self, prometheusservice_client):
+        workspace_alias = random_suffix_name("amp-workspace", 24) 
+        resource_name = random_suffix_name("rule-groups-namespace", 30)
+
+        # Create the workspace where the rule groups will be stored. 
+        workspace_id = self.create_workspace(prometheusservice_client, workspace_alias)
```

Comment:
How about using the same controller to manage (create/delete) the workspace resource? :) 
We have examples of tests doing this like `Aliases` and `Functions` in the lambda-controller https://github.com/aws-controllers-k8s/lambda-controller/blob/main/test/e2e/tests/test_alias.py#L82

Reply:
Good idea!